### PR TITLE
fix(ingest/snowflake): always ingest view and external table ddl lineage

### PR DESCRIPTION
**Pull request by mayurinehate**
Earlier disabling table lineage via `include_table_lineage` mandated disabling view lineage via `include_view_lineage` which made it impossible to separate schema and lineage/usage ingestions in separate recipes without missing on view lineage entirely.

Stacked on top of https://github.com/datahub-project/datahub/pull/12179

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py [hunk 1]
```diff
@@ -163,26 +163,13 @@ class SnowflakeConfig(
         default=True,
         description="If enabled, populates the snowflake table-to-table and s3-to-snowflake table lineage. Requires appropriate grants given to the role and Snowflake Enterprise Edition or above.",
```
Comment (sgomezvillamor) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py:164]: should we update this `include_table_lineage` description here to emphasize that this enables view lineage too?

**Review by sgomezvillamor — APPROVED** [review]
LGTM
Minor question in `include_table_lineage` param description

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py [hunk 2]
```diff
@@ -512,15 +513,14 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         discovered_datasets = discovered_tables + discovered_views
 
         if self.config.use_queries_v2:
-            self.report.set_ingestion_stage("*", "View Parsing")
-            assert self.aggregator is not None
+            self.report.set_ingestion_stage("*", VIEW_PARSING)
             yield from auto_workunit(self.aggregator.gen_metadata())
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py:517]: mostly for my own understanding - `self.aggregator` is only used for view parsing? and then there's another aggregator within `SnowflakeQueriesExtractor`?

if so, should the `self.aggregator.gen_metadata()` call happen first and then lineage generation happen afterwards?

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py:517]: > mostly for my own understanding - self.aggregator is only used for view parsing? and then there's another aggregator within SnowflakeQueriesExtractor?

Yes, when `use_queries_v2 is true`

When `use_queries_v2 is false`:
 - and `include_table_lineage is false` -> then self.aggregator is used for only view lineage
 - and `include_table_lineage is true` -> then self.aggregator is used for both view and table lineage.

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py:164]: The view upstream lineage would be ingested always irrespective of this flag.

**Pull request merged**